libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -73,10 +74,43 @@ bool ControlFlowBuilder::visit(BinaryOperation const& _operation)
 			return false;
 		}
 		default:
-			return ASTConstVisitor::visit(_operation);
+		{
+			visitNode(_operation);
+			if (_operation.annotation().userDefinedFunction)
+			{
+				solAssert(!m_currentNode->resolveFunctionCall(nullptr));
+				m_currentNode->functionCall = _operation.annotation().userDefinedFunction;
+
+				auto nextNode = newLabel();
+
+				connect(m_currentNode, nextNode);
+				m_currentNode = nextNode;
+			}
+			return true;
```

Comment:
If we return here, it means the children are visited after the node itself. Is this actually correct? Can you construct a test where the arguments to the operator and the operator itself have side-effects such that this breaks something?

as far as I can tell, the control flow graph is mainly used in ControlFlowAnalyzer to check for uninitialized access and unreachable code.

Reply:
I have updated one out of the tests we already have: https://github.com/ethereum/solidity/pull/12362/files#diff-a55f9c2c718b8eb3f967190622daa1a03b4146e2c018d9fa3dd44d9fb137f726. Please let me know if this is what you expected. To me it seems we get all expected warnings/errors so visiting the node before child nodes is not a problem here.

Reply:
I'm pretty sure that this is incorrect. This way, we have a node that calls the function that is then connected to a node that evaluates the arguments. Please take a look at how it is done for FunctionCall ("internal") and do it like it is done there.

Reply:
As agreed this is updated to "return ASTConstVisitor::visit(_operation);"